src/main/java/mate/hwdao/dao/impl/jdbc/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,197 @@
+            return Optional.ofNullable(car);
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't get car with " + id, ex);
+        }
+
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT * FROM cars INNER JOIN manufactures"
+                + " ON cars.manufacturer_id = manufactures.id WHERE deleted = false";
+        try (PreparedStatement preparedStatement
+                     = ConnectionUtil.getConnection()
+                .prepareStatement(query)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            List<Car> cars = new ArrayList<>();
+            while (resultSet.next()) {
+                cars.add(makeCar(resultSet));
+            }
+            return cars;
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't get list of cars ", ex);
+        }
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateCar = "UPDATE cars SET manufacturer_id = ?, model = ? "
+                + "WHERE id = ? AND deleted = false";
+        String deletedOldDrivers = "DELETE FROM cars_drivers WHERE car_id = ?";
+        String setNewDrivers = "INSERT INTO cars_drivers(driver_id, car_id) VALUES (?, ?)";
+        try (PreparedStatement updateCarStatement
+                     = ConnectionUtil.getConnection()
+                .prepareStatement(updateCar)) {
+            updateCarStatement.setLong(1, car.getManufacturer().getId());
+            updateCarStatement.setString(2, car.getModel());
+            updateCarStatement.setLong(3, car.getId());
+            updateCarStatement.executeUpdate();
+            delateOldDrivers(deletedOldDrivers, car);
+            setNewDriversStatement(setNewDrivers,car);
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't save " + car
+                    + " failed", ex);
+        }
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET deleted = true WHERE id = ?";
+        try (PreparedStatement preparedStatement
+                     = ConnectionUtil.getConnection()
+                .prepareStatement(query)) {
+            preparedStatement.setLong(1, id);
+            return preparedStatement.executeUpdate() > 0;
+        } catch (SQLException ex) {
+            throw new DataProcessingException("cars with " + id
+                    + "was not deleted because of", ex);
+        }
+    }
+
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "Select cars.id, manufacturer_id, model, deleted, "
+                + "manufactures.id, manufactures.name, manufactures.country "
+                + "From cars  INNER JOIN manufactures ON cars.manufacturer_id = manufactures.id "
+                + "JOIN cars_drivers WHERE cars.id = cars_drivers.car_id "
+                + "AND  cars_drivers.driver_id = ? AND deleted = false";
```

Comment:
```suggestion
        String query = "SELECT c.id as car_id, model, "
                + "m.id as manufacturer_id, name, country "
                + "FROM cars c INNER JOIN manufactures m ON c.manufacturer_id = m.id "
                + "JOIN cars_drivers cd ON c.id = cd.car_id "
                // join drivers d
                + "WHERE cd.driver_id = ? AND c.deleted = false AND d.deleted = false";
```

Reply:
ok, I added check for driver deleted, but I nothing understand when you use short name. I left my old naming